clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -97,8 +101,21 @@ public FileOutStream(long fileId, OutStreamOptions options) throws IOException {
     }
     mClosed = false;
     mCanceled = false;
-    mHostname = options.getHostname();
     mShouldCacheCurrentBlock = mTachyonStorageType.isStore();
+
+    // get default policy
+    Class<? extends FileWriteLocationPolicy> locationPolicyClass = ClientContext.getConf()
+        .<FileWriteLocationPolicy>getClass(Constants.USER_FILE_WRITE_LOCATION_POLICY);
+    if (options.getLocationPolicyClass() != null) {
+      locationPolicyClass = options.getLocationPolicyClass();
+    }
+    try {
+      mLocationPolicy = LocationPolicyRegistry.create(locationPolicyClass,
```

Comment:
so regardless of location policy, we always need to `getBlockWorkerInfoList()` (i.e., a client-master RPC)?
it sounds to me sometimes unnecessary, isn't it?